### ISSUE TITLE
When use count++ -  symbol recognized as  is never read and symbol greyed out when checkJS enabled

### ISSUE THREAD

**Issue by lukaszpolowczyk** [Microsoft/TypeScript#30177]
Steps to Reproduce:

1. Enable checkJS
2. Use `count++;` or `count += 1;`
![count-bug](https://user-images.githubusercontent.com/16800535/53629874-e9216780-3c0e-11e9-8a9a-7e86b7b07d64.png)
3. Info: 'count' is declared but its value is never read.ts(6133)
And symbol is greyed out.

it works well `count = count+1;`
![count](https://user-images.githubusercontent.com/16800535/53629897-fb030a80-3c0e-11e9-95e9-98763118a48f.png)

**Comment by mjbvz** [comment]
This warning make sense since the variable is never used (only updated). The bug is that the `count = count+1;`  case should also show this warning

**Comment by lukaszpolowczyk** [comment]
@mjbvz And when I do, for example, `number = number + Math.random()`, this is also only an update? When is it not an update?

Maybe offtop, but to the topic.

**Comment by mjbvz** [comment]
Yes, you are only updating the value of the var and never using it for anything. It could be considered unused

**Comment by lukaszpolowczyk** [comment]
@mjbvz I do not understand this.  Provide examples where the variable is actually used.

**Comment by mjbvz** [comment]
```js
let count = 0;
++count;
console.log(count); // Used
```

**Comment by lukaszpolowczyk** [comment]
@mjbvz  Passing as an argument to another function?
Actually, in a sense, the `+` operator can also be interpreted as a function with two argumets.

But to the subject: Graying should rather disappear when you do `++` or `+=`, because the sense of graying intuitively is just that.

**Comment by mjbvz** [comment]
If an operation has no side effects, such as adding two numbers together and then never using the result, it may as well never have happened. Having such code in your program is very likely a programming error.

Your examples increment a var and then do nothing with the result. The var is never used. If you deleted it, the program would produce the exact same output

**Comment by RyanCavanaugh** [comment]
"Used *except* in a side-effect-free expression which is ultimately only assigned to itself" isn't a level of analysis we're interested in pursuing for the sake of what is effectively a lint rule.

**Comment by fenderOne** [comment]
**So**,

**count++** is not the same as **count = count + 1**?

This grayoff's "two different behaviors" for the same result cost me many hours looking for an error!!
I think it should grayoff in both cases! (consistency)
